### PR TITLE
Add Github workflow to check _psl DNS entries on PRs

### PR DESCRIPTION
**Pull request by simon-friedberger**
Let's have more automatic checks!

### CODE REVIEW

**Comment by weppos** [comment]
Hi @simon-friedberger, can you provide a bit more context about this PR. Is there a conversation somewhere about it? What's the purpose of this script, and what is the expectation?

I also wonder, I see it was developed in Python. Is it because that's the language you are most familiar with, or is there another reason? I am asking as we had Python code in the past but it was not really the most preferred language and we moved most of the tooling to Go. I am a bit concerned about adding more Python again, for the long term maintainability.

**Comment by dnsguru** [comment]
@simone I asked simon to pause this...  There is a separate dialog with
google about development of automation which I believe was go-based - I
have shared the diagrams and specs I took time to develop in the hopes of
avoiding parallel evolution costing valuable volunteer cycles

**Comment by simon-friedberger** [comment]
I am not against using Go for this if that is the project preference. Python was just the natural choice for me.

**Comment by dnsguru** [comment]
> I am not against using Go for this if that is the project preference. Python was just the natural choice for me.

Same thought, as go is less familiar to me, but want to minimize complexity with >1 approaches at this point so we don't add interop burdens

**Comment by simon-friedberger** [comment]
The submission by Sam [here](https://github.com/publicsuffix/list/pull/1953) is also in Python and @dnsguru has expressed support. Can I go ahead with this?

**Comment by dnsguru** [comment]
After discussion, this looks like a good idea - accompanied by test suite elements to know about impacts.  The automation work is really appreciated

File: tools/pr_checker/check_dns.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import dns.query
+import dns.resolver
+import re
+
+
+def read_rules(psl_filename):
+    """Read rules from a file into a set.
+
+    >>> sorted(read_rules('test_psl.dat'))
+    ['!main.bar.example.org', '*.bar.example.org', 'a.example.com', 'b.example.com.', 'example.com', 'example.org', 'foo.example.org']
```
Comment (dnsguru) [inline tools/pr_checker/check_dns.py:13]: do we need a flat TLD and flat TLD w wildcard rule or not here?

ex: add ```'.com', '*.com'```

**Comment by dnsguru** [comment]
I suggested a TLD and TLD w a sld wildcard in the examples for the test suite 

I was also thinking of IDN support.  Upon some contemplation, I believe those scenarios would funciton identically to their LDH equivelents in the test suite.  Just documenting here that it was considered.  Adding them here would be placebo; cosmetic-only, require additional dependencies/libraries/modules, add complexity and not have any difference in outcome, so would not deliver any value or benefit in exchange for the overhead.

**Comment by simon-friedberger** [comment]
As soon as I get to it we should merge this with parsing and checks for individual rules in the linter here https://github.com/publicsuffix/list/blob/master/linter/pslint.py that should cover IDN issues.

**Comment by simon-friedberger** [comment]
@weppos Please have a look!

File: tools/pr_checker/check_dns.py (hunk 1 above)
Comment (simon-friedberger) [reply tools/pr_checker/check_dns.py]: Done!

**Pull request merged**